client/src/axiom/utils.ts

```diff
@@ -59,4 +65,32 @@ export function argsObjToArr(
     args.refundee,
     args.dataQuery,
   ]
+}
+
+export async function getMaxFeePerGas(axiom: AxiomSdkCore, maximumMaxFeePerGas?: string): Promise<string> {
+  if (maximumMaxFeePerGas === undefined) {
+    maximumMaxFeePerGas = ClientConstants.MAX_MAX_FEE_PER_GAS.toString();
+  }
+  const providerFeeData = (await axiom.config.provider.getFeeData()).maxFeePerGas as bigint;
+  const publicClient = createPublicClient({
+    chain: convertChainIdToViemChain(axiom.config.chainId.toString()),
+    transport: http(axiom.config.providerUri),
+  });
+  let contractMinMaxFeePerGas = await publicClient.readContract({
+    address: axiom.getAxiomQueryAddress() as `0x${string}`,
+    abi: axiom.getAxiomQueryAbi(),
+    functionName: "minMaxFeePerGas",
+    args: [],
+  }) as bigint;
+  if (contractMinMaxFeePerGas === 0n) {
+    contractMinMaxFeePerGas = ClientConstants.MIN_MAX_FEE_PER_GAS;
+  }
+  if (providerFeeData > contractMinMaxFeePerGas) {
+    const maximumMaxFeePerGasBigInt = BigInt(maximumMaxFeePerGas);
+    if (providerFeeData > maximumMaxFeePerGasBigInt) {
```

Comment:
In these cases we should have a descriptive console.log saying the fee change we made, and how it can be overridden with options.